internal/provider/database/resource.go

```diff
@@ -64,7 +76,44 @@ func (state *DatabaseResourceModel) Read(ctx context.Context, client openapi.Cli
 	return helper.ParseResponse(resp, state)
 }
 
-func (state *DatabaseResourceModel) Update(ctx context.Context, client openapi.ClientInterface) error {
+const (
+	DBA_PASSWORD_CHANGE_UNSUPPORTED_MSG = "Configured DBA password was changed and the server does not support updating the DBA password. " +
+		"Revert the configured DBA password to the value in the Terraform state and retry."
+)
+
+func IsDbaPasswordUpdateUnsupportedError(resp *http.Response, err error) bool {
+	if err != nil {
+		// "404 Not Found" is returned with no "detail" message if /dbaPassword	sub-resource is not supported
+		if resp.StatusCode == http.StatusNotFound {
+			if apiError, ok := err.(*helper.ApiError); !ok || apiError.GetDetail() == "" {
+				return true
+			}
+		}
+	}
+	return false
+}
```

Comment:
If I am not mistaken, this check is going to be the same whenever we try to access a URL that the configured CP does not understand. Presumably, this will happen again whenever we add any new APIs. Why not give it a generic name and put it in a common file?

Reply:
There are multiple response status codes that can be associated with an unknown request. If the resource path is valid, but the request method is unknown, then "405 Method Not Allowed" is returned. There is also the "415 Unsupported Media Type" response, which may be due to an unknown `Content-Type` header being supplied.

Either way, it does make sense to generalize the the name and perhaps cover all of the possible cases, but if we get 2.5.0 out quickly enough, this will be the only request that this check will be needed for, and the new `UNKNOWN_REQUEST` API code can be used to check support for all future requests.

Reply:
I will not generalize this since I am not sure that it is the case that every single "404 Not Found" with an empty "detail" message is the result of request dispatching error. I know that it is the case for `/dbaPassword`, since the only other scenario where it throws "404 Not Found" is when the database not exist and a "No database named ..." message appears in the "detail" field.